### PR TITLE
Support MSC3375: room version 9.

### PR DESCRIPTION
**Pull request by clokep**
Per [MSC3375](https://github.com/matrix-org/matrix-doc/pull/3375).

This needs to wait for that the MSC to be merged.

### CODE REVIEW

File: synapse/api/room_versions.py [hunk 1]
```diff
@@ -197,12 +207,29 @@ class RoomVersions:
...
         msc2176_redaction_rules=False,
         msc3083_join_rules=False,
+        msc3375_redaction_rules=False,
         msc2403_knocking=True,
         msc2716_historical=False,
         msc2716_redactions=False,
     )
     V8 = RoomVersion(
         "8",
+        RoomDisposition.UNSTABLE,
```
Comment (clokep) [inline synapse/api/room_versions.py:217]:
```suggestion
        RoomDisposition.STABLE,
```

File: synapse/api/room_versions.py [hunk 2]
```diff
@@ -293,7 +324,7 @@ class RoomVersionCapability:
         ),
         RoomVersionCapability(
             "restricted",
-            RoomVersions.V8,
+            RoomVersions.V9,
```
Comment (clokep) [inline synapse/api/room_versions.py:327]: We probably don't want to do this at the same time, we might want to ensure we get some people upgraded first.

Comment (clokep) [reply synapse/api/room_versions.py:327]: My initial feeling is we should probably wait 1 version before doing this, but would be curious of other's thoughts.

Comment (erikjohnston) [reply synapse/api/room_versions.py:327]: I think it depends on how the clients are representing this flag to the user. If we enable it then obviously at first very few other servers will be able to join the newly created rooms, which might be fine if the UI makes it clear its experimental?

Comment (clokep) [reply synapse/api/room_versions.py:327]: It uses this to determine what room version to upgrade to if attempting to change a room to use restricted join rules for an old room version. This leads me to think we should wait until there's at least some uptake before doing this? 🤷 

image

**Review by erikjohnston — APPROVED** [review]

File: synapse/api/room_versions.py (hunk 2 above)
Comment (clokep) [reply synapse/api/room_versions.py:327]: After some discussion the decision was to change the preferred version 1 release after v9 is available.

Comment (clokep) [reply synapse/api/room_versions.py:327]:
```suggestion
            RoomVersions.V8,
```

**Pull request merged**